### ISSUE TITLE
Wrong subscription id

### ISSUE THREAD

**Issue by andreasgerstmayr** [microsoft/azserialconsole#83]
I'm getting the following when opening the serial console:
```
The serial console connection to the VM encountered an error: 'Forbidden'
 (SubscriptionNotEnabled) - Subscription name undefined, id XXXXXXXXX is in non-Enabled state undefined.
```

The shown subscription id (blanked out here by myself) is not my current subscription id. In the subscriptions list at https://portal.azure.com/#blade/Microsoft_Azure_Billing/SubscriptionsBlade I can only see one subscription, with a different subscription id.

**Comment by asinn826** [comment]
Hey @andreasgerstmayr - could you try the steps to disable and re-enable your subscription for serial console here? https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/serial-console-enable-disable

Ensure that the subscription you are disabling and re-enabling is the same subscription that contains the VM you're using serial console from.

**Comment by woolensculpture** [comment]
I am running into the same issue, disabling and reenabling serial console in the VM's subscription has not helped.

**Comment by andreasgerstmayr** [comment]
@asinn826: The subscription, which the serial console thinks I'm using, doesn't exist (anymore, I guess it was the Azure trial).

My *actual* subscription can disable and enable the serial console as shown in https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/serial-console-enable-disable
If I'm using the subscription id which the serial console shows in the error message, it shows `The subscription 'XXXX' could not be found.`

I only have **one** active subscription, but the serial console is showing another subscription (I guess the Azure trial one, which is deleted by now).

**Comment by asinn826** [comment]
@andreasgerstmayr - so to clarify - you were trying to access serial console on a VM that was created within an Azure trial subscription, that has expired now? I'm thinking that might be the expected behavior since upon creation, we would look for whether serial console is enabled for your trial subscription, but then the subscription would be in some deleted state that is unrecognizable to serial console, so we fail and report that forbidden error you originally saw. Are you seeing similar errors in other, active subscriptions? I know that wonky things can happen once trial subscriptions expire or are deleted and maybe this behavior falls in that category of wonky things.

@woolensculpture - were you in a similar situation as @andreasgerstmayr where you were using a trial subscription? Or did you see this error in a VM in an active subscription? Any more details would be helpful.

**Comment by andreasgerstmayr** [comment]
> @andreasgerstmayr - so to clarify - you were trying to access serial console on a VM that was created within an Azure trial subscription, that has expired now?

No, that VM was created with my current subscription. It happens on all newly created VMs, and right now I only have one active subscription.

> Are you seeing similar errors in other, active subscriptions?
No, I'm not aware of any.

Idk, maybe it's some error with my account? (still some bug nevertheless)
Should I contact you in private with the subscription details?

**Comment by woolensculpture** [comment]
no my situation was the subscription associated with the id was migrated to another AD domain. I have actually worked around this issue by granting an account in the  with access to the migrated subscription guest ownership permissions to both the VM and the storage account that the boot diagnostics are feeding into. I do not understand though why the console would need access to this subscription as it was related to a totally different system and neither the VM or storage account are under the migrated subscription.

**Comment by asinn826** [comment]
@andreasgerstmayr , @woolensculpture - we've done some investigation into this, and I think we're on the right track to hunt down this error. We suspect there's something going on with our logic that interacts with Cloud Shell but that's as much as I know right now. Hopefully we can get this resovled for you guys and everyone else that's seeing this.

**Comment by asinn826** [comment]
@andreasgerstmayr , @woolensculpture - can you guys share the subscription ids in which you saw this error? If you're not comfortable sharing in Github, feel free to email me at alfred.sin [at] microsoft.com.

**Comment by asinn826** [comment]
One more thought, which may address what you saw, @woolensculpture - if the storage account that you have associated with Cloud Shell is in a subscription that has been deprovisioned or deactivated for any reason, then you would see this error. So to mitigate, you would have to change the storage account associated with Cloud Shell to use a valid subscription, or delete Cloud Shell storage entirely. That should resolve the error, which is what I think you may have seen as well.
